plugin/core/sessions.py

```diff
@@ -1642,7 +1642,19 @@ def open_uri_async(
         # There is no pre-existing session-buffer, so we have to go through AbstractPlugin.on_open_uri_async.
         if self._plugin:
             return self._open_uri_with_plugin_async(self._plugin, uri, r, flags, group)
-        return Promise.resolve(None)
+        return None
+
+    def open_uri_async(
+        self,
+        uri: DocumentUri,
+        r: Optional[Range] = None,
+        flags: int = 0,
+        group: int = -1
+    ) -> Promise[Optional[sublime.View]]:
+        promise = self.try_open_uri_async(uri, r, flags, group)
+        if promise is None:
+            raise RuntimeError("unexpected URI scheme")
```

Comment:
There is code like https://github.com/sublimelsp/LSP/blob/07ef2a05e55034a222e6f2f4da13b463a51ac417/plugin/locationpicker.py#L31-L35 and https://github.com/sublimelsp/LSP/blob/757908ad385d22d31c3dd69c6a898fcb50758571/plugin/core/registry.py#L243-L250 that did show sublime message on unhandled schema and I guess it will no longer trigger and we'll get an exception instead.

I must say that I feel a bit weird about this returning either `None` or `Promise[None]`. Would be more consistent with always being a Promise like before...

Reply:
This is a good point, I'll have to study this again.

Reply:
I restored the original behavior.